File: frontend/src/services/commandTemplate.ts
import api from '../config/api';

export interface CommandTemplate {
  id: number;
  name: string;
  description?: string;
  category?: 'system' | 'network' | 'disk' | 'process' | 'custom';
  command_function: string;
  command_args?: any[];
  icon?: string;
  created_by: number;
  is_public: boolean;
  usage_count: number;
  created_at: string;
  updated_at: string;
  created_by_user?: {
    id: number;
    username: string;
    display_name?: string;
  };
}

export interface CreateCommandTemplateRequest {
  name: string;
  description?: string;
  category?: 'system' | 'network' | 'disk' | 'process' | 'custom';
  command_function: string;
  command_args?: any[];
  icon?: string;
  is_public?: boolean;
}

export interface UpdateCommandTemplateRequest {
  name?: string;
  description?: string;
  category?: 'system' | 'network' | 'disk' | 'process' | 'custom';
  command_function?: string;
  command_args?: any[];
  icon?: string;
  is_public?: boolean;
}

export const commandTemplateService = {
  // 创建命令模板
  createTemplate: async (data: CreateCommandTemplateRequest): Promise<{ template: CommandTemplate }> => {
    const response = await api.post('/command-templates', data);
    return response.data;
  },

  // 列出命令模板
  listTemplates: async (filters?: { category?: string; is_public?: boolean }): Promise<{ templates: CommandTemplate[] }> => {
    const response = await api.get('/command-templates', {
      params: filters,
    });
    return response.data;
  },

  // 获取模板详情
  getTemplate: async (id: number): Promise<{ template: CommandTemplate }> => {
    const response = await api.get(`/command-templates/${id}`);
    return response.data;
  },

  // 更新模板
  updateTemplate: async (id: number, data: UpdateCommandTemplateRequest): Promise<void> => {
    await api.put(`/command-templates/${id}`, data);
  },

  // 删除模板
  deleteTemplate: async (id: number): Promise<void> => {
    await api.delete(`/command-templates/${id}`);
  },
};

File: backend/internal/models/command_template.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

// CommandTemplate 命令模板模型
type CommandTemplate struct {
	ID              uint           `gorm:"primaryKey" json:"id"`
	Name            string         `gorm:"size:255;not null" json:"name"`
	Description     string         `gorm:"type:text" json:"description"`
	Category        string         `gorm:"size:50;index" json:"category"` // system, network, disk, process, custom
	CommandFunction string         `gorm:"size:255;not null" json:"command_function"`
	CommandArgs     datatypes.JSON `gorm:"type:jsonb" json:"command_args"`
	Icon            string         `gorm:"size:50" json:"icon"`
	CreatedBy       uint64         `gorm:"not null;index" json:"created_by"`
	IsPublic        bool           `gorm:"default:false" json:"is_public"`
	UsageCount      int            `gorm:"default:0" json:"usage_count"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`

	// 关联
	CreatedByUser User `gorm:"foreignKey:CreatedBy" json:"created_by_user,omitempty"`
}

// TableName 指定表名
func (CommandTemplate) TableName() string {
	return "command_templates"
}

